CONTRIBUTING.md

```diff
@@ -153,6 +154,89 @@ The latest dependencies generated:
 You can also see dependency cycles in case there are some cycles in
 [o2a-dependency-cycles.png](images/o2a-dependecy-cycles.png)
 
+# Continuous integration environment
+
+The project integrates with Travis CI. To enable saving of the build process artifacts, you must configure the authorization mechanisms for Google Cloud Storage.
+
+1. To simplify the instructions, set the environment variable:
+```bash
+PROJECT_ID="$(gcloud config get-value project)"
+ACCOUNT_NAME=o2a-build-artifacts-travis
+ACCOUNT_EMAIL="${ACCOUNT_NAME}@${PROJECT_ID}.iam.gserviceaccount.com"
+BUCKET_NAME=o2a-build-artifacts
+```
+
+2. Create the service account that will be used by Travis
+```bash
+gcloud iam service-accounts create "${ACCOUNT_NAME}"
+```
+
+3. Create a new private key for the service account, and save a copy of it in the `.travis/o2a-build-artifacts-sa.json` file. This file is used by Travis CI to authorize action.
+```bash
+gcloud iam service-accounts keys create --iam-account "${ACCOUNT_EMAIL}" .travis/o2a-build-artifacts-sa.json
+```
+
+4. Create the bucket
+```bash
+gsutil mb "gs://${BUCKET_NAME}"
+```
+
+5. Enables the Bucket Policy Only feature on Cloud Storage bucket(s)
+```bash
+gsutil bucketpolicyonly set on "gs://${BUCKET_NAME}"
+```
+
+6. Grant permission to make a bucket's objects publicly readable:
+```bash
+gsutil iam ch allUsers:objectViewer "gs://${BUCKET_NAME}"
+```
+
+7. Grant permission to create and overwrite a bucket's objects by CI service account:
+```bash
+gsutil iam ch user:${ACCOUNT_EMAIL}:objectAdmin gs://ex-bucket
+```
+
+8. Encrypt file using [Travis CLI](https://github.com/travis-ci/travis.rb). See more: [Encrypting Files on Travis CI](https://docs.travis-ci.com/user/encrypting-files/).
+```bash
+travis encrypt-file .travis/o2a-build-artifacts-sa.json
+```
+You should see a result similar to the one below
+```
+encrypting .travis/o2a-build-artifacts-sa.json for GoogleCloudPlatform/oozie-to-airflow
+storing result as a.txt.enc
+storing secure env variables for decryption
+
+Please add the following to your build script (before_install stage in your .travis.yml, for instance):
+
+    openssl aes-256-cbc -K $encrypted_9d13a0bd74e6_key -iv $encrypted_9d13a0bd74e6_iv -in a.txt.enc -out a.txt -d
+
+Pro Tip: You can add it automatically by running with --add.
+```
+
+9. Write down the variable ID with the decryption key:
+```bash
+export TRAVIS_VAR_ID=9d13a0bd74e6
+```
+
+10. Update a travis.yml with new variables:
```

Comment:
The problem with this is that you will have to add your own key in .travis.yml - which is shared between forks and you do not want to have a commit with name of the variable changed that you want to move together with your code in your fork. We have to figure out how to pass the name of encryption key via environment variable rather than by modifying travis.yml. This should not be difficult via indirection: http://mywiki.wooledge.org/BashFAQ/006#Indirection